runtime/src/hardened_unpack.rs

```diff
@@ -337,10 +343,61 @@ pub fn unpack_snapshot<A: Read>(
                 UnpackPath::Invalid
             }
         },
+        |_| {},
     )
     .map(|_| unpacked_append_vec_map)
 }
 
+pub fn streaming_unpack_snapshot<A: Read>(
+    archive: &mut Archive<A>,
+    ledger_dir: &Path,
+    account_paths: &[PathBuf],
+    parallel_selector: Option<ParallelSelector>,
+    sender: &crossbeam_channel::Sender<PathBuf>,
+) {
+    let mut i = 0;
+    unpack_archive(
+        archive,
+        MAX_SNAPSHOT_ARCHIVE_UNPACKED_APPARENT_SIZE,
+        MAX_SNAPSHOT_ARCHIVE_UNPACKED_ACTUAL_SIZE,
+        MAX_SNAPSHOT_ARCHIVE_UNPACKED_COUNT,
+        |parts, kind| {
+            if is_valid_snapshot_archive_entry(parts, kind) {
+                i += 1;
+                match &parallel_selector {
+                    Some(parallel_selector) => {
+                        if !parallel_selector.select_index(i - 1) {
+                            return UnpackPath::Ignore;
+                        }
+                    }
+                    None => {}
+                };
+                if let ["accounts", _] = parts {
+                    // Randomly distribute the accounts files about the available `account_paths`,
+                    let path_index = thread_rng().gen_range(0, account_paths.len());
+                    match account_paths
+                        .get(path_index)
+                        .map(|path_buf| path_buf.as_path())
+                    {
+                        Some(path) => UnpackPath::Valid(path),
```

Comment:
I'm confused why this copies a lot of code from hardened_unpack.rs. I think right here is the main/only difference? I imagine this can be combined.